src/components/dls/Modal/Modal.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { useCallback, useRef, ReactNode, useEffect } from 'react';
+import useKeyPressedDetector from 'src/hooks/useKeyPressedDetector';
+import useOutsideClickDetector from 'src/hooks/useOutsideClickDetector';
+import styled from 'styled-components';
+import Close from '../../../../public/icons/close.svg';
+import Button, { ButtonSize } from '../Button/Button';
+
+interface Props {
```

Comment:
I tried out the modal locally. Overall it works really well masha'Allah. I noticed a couple of things: 

1) the x button is not scaled correctly:

<img width="1203" alt="2021-07-24 at 14 29 16@2x" src="https://user-images.githubusercontent.com/11590314/126868492-ea9bd8ea-c699-48a5-920d-9154cdb4bed4.png">

2) The vertically scrolling and horizontally scrolling modals should not behave the way they do. There should be a max width specified by `size` prop, and there should be a max height (perhaps a % of the viewport). If the children size exceeds the modal size, they should overflow (scroll within the container). This is the effect when I try to overflow the gestalt (pinterest design system library) modal: 

<img width="578" alt="2021-07-24 at 14 36 38@2x" src="https://user-images.githubusercontent.com/11590314/126868650-717ead3c-4357-45da-91b4-f0b71194fd97.png">

Reply:
Regarding 1st point it's an issue with Storybook and `Button` component not with `Modal` itself. The button looks fine in the UI but it's shifted a bit when it's viewed as a story.

<img width="707" alt="Screen Shot 2021-07-25 at 08 24 44" src="https://user-images.githubusercontent.com/15169499/126887195-ad8adcd9-2a97-4b5c-9728-1f2bcbafe72a.png">

This is how the close button for the modal looks without Storybook: 

<img width="611" alt="Screen Shot 2021-07-25 at 08 26 40" src="https://user-images.githubusercontent.com/15169499/126887213-ff168a4a-4ae3-4c51-8b8c-af71f6f2f8df.png">